src/Protocol/Protocol_1_9.cpp

```diff
@@ -2384,7 +2384,7 @@ void cProtocol_1_9_0::HandlePacketBlockPlace(cByteBuffer & a_ByteBuffer)
 	HANDLE_READ(a_ByteBuffer, ReadBEUInt8, UInt8, CursorX);
 	HANDLE_READ(a_ByteBuffer, ReadBEUInt8, UInt8, CursorY);
 	HANDLE_READ(a_ByteBuffer, ReadBEUInt8, UInt8, CursorZ);
-	m_Client->HandleRightClick(BlockX, BlockY, BlockZ, FaceIntToBlockFace(Face), CursorX, CursorY, CursorZ, m_Client->GetPlayer()->GetEquippedItem());
+	m_Client->HandleRightClick(BlockX, BlockY, BlockZ, FaceIntToBlockFace(Face), CursorX, CursorY, CursorZ, Hand);
```

Comment:
Since we convert the block face to an enum here via `FaceIntToBlockFace`, I would simply do the same for the `Hand` parameter. Then you can also change the function signature to `eHand`.
P.S.: I would use a switch-case instead of an `static_cast`. Then the enum isn't required to match the protocol.

Reply:
Done